### DISCUSSION TITLE
GitHub Action Failing on Older GitHub Enterprise Version Due to Node Version Compatibility

### DISCUSSION THREAD

**Discussion by sadmananik** [community/community#123591] [Enterprise]
**Issue Description**

We are encountering failures with GitHub Actions on a self-hosted runner running GitHub Enterprise Server version 3.4.1, where the latest available action runner version is 2.285.1. The problem stems from compatibility issues with node versions required by certain plugin actions. Specifically, the plugins we are using require node version 20, which is not available or supported on our GitHub Enterprise instance.

Request for Multiple Node Version Support in Plugin Action.yml File:

I'm seeking advice or a solution to specify multiple node versions within the action.yml file of a plugin. This would enable compatibility with older GitHub Enterprise versions like 3.4.1, which do not support the latest node versions required by newer plugin actions.

Any assistance or workaround suggestions to resolve this node version conflict on our self-hosted runner would be greatly appreciated.

Thank you for your attention to this matter.

**Comment by jsoref** [comment]
Fwiw, I put together a thing that lets you rewrite the node version used by actions: https://github.com/check-spelling-sandbox/jubilant-journey -- it's a proof-of-concept as opposed to production code, but I expect people will want something again as projects again rush to drop a node version and fail to offer support for two...
